### PR TITLE
Add missing overloads to getComponents

### PR DESCRIPTION
**Pull request by rafern**
`Object3D.getComponent` correctly infers the type of a native component from the name of the component. `Object3D.getComponents` doesn't.

### CODE REVIEW

**Comment by DavidPeicho** [comment]
Sorry for the delay on this. I will review that ASAP

File: src/wonderland.ts [hunk 1]
```diff
@@ -4125,6 +4125,29 @@ export class Object3D {
...
+    /** @overload */
+    getComponents(type: 'light'): LightComponent[];
+    /** @overload */
+    getComponents(type: 'animation'): AnimationComponent[];
+    /** @overload */
+    getComponents(type: 'physx'): PhysXComponent[];
+    /** @overload */
+    getComponents(type?: string | null): Component[];
+    /** @overload */
+    getComponents<T extends Component>(clazz: ComponentConstructor<T>): T[];
```
Comment (DavidPeicho) [inline src/wonderland.ts:4149]: Instead of `clazz` please use something like `typeClass`

**Comment by DavidPeicho** [comment]
Except for the minor typo, happy to merge!

File: src/wonderland.ts (hunk 1 above)
Comment (rafern) [reply src/wonderland.ts]: Done

File: src/wonderland.ts [hunk 2]
```diff
@@ -4125,6 +4125,29 @@ export class Object3D {
...
+    /** @overload */
+    getComponents(type: 'animation'): AnimationComponent[];
+    /** @overload */
+    getComponents(type: 'physx'): PhysXComponent[];
+    /** @overload */
+    getComponents(type?: string | null): Component[];
+    /** @overload */
+    getComponents<T extends Component>(typeClass: ComponentConstructor<T>): T[];
+
     /**
```
Comment (DavidPeicho) [inline src/wonderland.ts:4151]: Do you mind moving this comment to the above overload? Turns out typedoc doesn't parse comment on the implementation :(

File: src/wonderland.ts [hunk 3]
```diff
@@ -4659,7 +4679,10 @@ export class Object3D {
      * @note As this function is non-trivial, avoid using it in `update()` repeatedly,
      *      but rather store its result in `init()` or `start()`
      * @warning This method will currently return at most 341 components.
+     * @overload
      */
+    getComponents<T extends Component>(typeClass: ComponentConstructor<T>): T[];
```
Comment (DavidPeicho) [inline src/wonderland.ts:4684]: Can you rename that to `typeOrClass` please to match the param

**Comment by DavidPeicho** [comment]
Thanks a lot!

**Pull request merged**